<a href="https://colab.research.google.com/github/valentinabc19/Proyecto1-Estadistica1/blob/main/ProyectoAED.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**INTEGRANTES**

*Valentina Bueno Collazos - 2230556*

*Juan Pablo Gomez Veira - 2231715*

*Natalia Moreno Montoya - 2230434*

# Minnesota's DataFrame

In [ ]:
import pandas as pd
# Se crea el dataframe de minnesota desde el CSV en el repositorio de github
mn_df = pd.read_csv("https://raw.githubusercontent.com/valentinabc19/Proyecto1-Estadistica1/main/samples/minnesota_sample.csv")

# Se seleccionan solo las columnas de interés
mn_df = mn_df[['Region','Gender','Handed', 'Favourite_physical_activity', 'Importance_reducing_pollution',
               'Birth_month', 'Beverage', 'Favorite_School_Subject', 'Sleep_Hours_Non_Schoolnight',
               'Paid_Work_Hours', 'Work_At_Home_Hours','Planned_Education_Level']]

# Mostrar el DataFrame
mn_df
len(mn_df) #Verificación de total de filas

500

In [ ]:
mn_df = mn_df.dropna() #Se eliminan las filas con valores nulos
len(mn_df) #Verificación de total de filas restantes
#mn_df

339

In [ ]:
# Convierte la columna 'Importance_reducing_pollution' a numérico y maneja los valores no numéricos de manera adecuada
mn_df['Importance_reducing_pollution'] = pd.to_numeric(mn_df['Importance_reducing_pollution'], errors='coerce').astype(int)

# Elimina los valores que no están dentro del rango válido (0 a 1000)
mn_df = mn_df[(mn_df['Importance_reducing_pollution'] >= 0) & (mn_df['Importance_reducing_pollution'] <= 1000)]

mn_df[['Importance_reducing_pollution']]
len(mn_df)


<ipython-input-4-8e550d7a1319>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  mn_df['Importance_reducing_pollution'] = pd.to_numeric(mn_df['Importance_reducing_pollution'], errors='coerce').astype(int)


339

In [ ]:
"""
Se limpia específicamente la columna "Sleep_Hours_Non_Schoolnight" ya que los datos que esta contiene
son digitados manualmente por los usuarios, por lo que es más susceptible a tener valores indeseados
"""
import numpy as np

# Función para limpiar la columna 'Sleep_Hours_Non_Schoolnight'
def clean_sleep_hours(value):
    try:
        # Elimina cualquier carácter no numérico y espacios en blanco
        value = value.strip().lower().replace('hrs', '').replace(' ', '')

        if '-' in value:  # Convierte intervalos en sus promedios
            start, end = map(float, value.split('-'))
            return round((start + end) / 2)
        elif value.isdigit() or value.replace('.', '', 1).isdigit():  # Maneja números y decimales
            value = float(value)
            if 0 <= value <= 24:  # Considera valores dentro de un rango realista para horas de sueño
                return int(round(value))
        return np.nan  # Asigna NaN a valores inválidos o atípicos
    except:
        return np.nan  # Maneja cualquier caso inesperado

# Aplica la función de limpieza a la columna
mn_df['Sleep_Hours_Non_Schoolnight'] = mn_df['Sleep_Hours_Non_Schoolnight'].apply(lambda x: clean_sleep_hours(str(x)))

# Elimina las filas con NaN en 'Sleep_Hours_Non_Schoolnight' (valores atípicos y entradas inválidas)
mn_df = mn_df.dropna(subset=['Sleep_Hours_Non_Schoolnight'])

# Convierte la columna 'Sleep_Hours_Non_Schoolnight' a tipo entero
mn_df['Sleep_Hours_Non_Schoolnight'] = mn_df['Sleep_Hours_Non_Schoolnight'].astype(int)

# Muestra el dataframe limpio
mn_df[['Sleep_Hours_Non_Schoolnight']]
mn_df

Region  Gender        Handed Favourite_physical_activity  \
0       MN  Female  Right-Handed                Hockey (Ice)   
1       MN    Male  Right-Handed              Hockey (Field)   
2       MN    Male  Right-Handed           Baseball/Softball   
4       MN  Female  Right-Handed                Martial Arts   
5       MN    Male  Right-Handed                       Other   
..     ...     ...           ...                         ...   
489     MN  Female  Right-Handed                     Dancing   
490     MN  Female  Right-Handed                    Swimming   
493     MN    Male  Right-Handed         Football (American)   
494     MN  Female  Right-Handed                    Swimming   
495     MN  Female  Right-Handed                      Tennis   

     Importance_reducing_pollution Birth_month                  Beverage  \
0                              316     October                     Water   
1                              690        July                     Water   
2                             1000     October                     Water   
4                              999    February                     Water   
5                              396     January            Powdered drink   
..                             ...         ...                       ...   
489                            800     October                       Tea   
490                            519    February                       Tea   
493                            675   September  Soft drink (caffeinated)   
494                            900         May                     Water   
495                            765        June                     Water   

    Favorite_School_Subject  Sleep_Hours_Non_Schoolnight Paid_Work_Hours  \
0                   English                           10              20   
1                   Science                            8              15   
2        Physical education                           12               2   
4                       Art                            5              40   
5        Physical education                           14               0   
..                      ...                          ...             ...   
489                 Science                            9               0   
490                 Science                            4               0   
493      Physical education                           10              10   
494                 Science                           11              10   
495                   Music                            9              12   

    Work_At_Home_Hours Planned_Education_Level  
0                    0         Graduate degree  
1                    8   Less than high school  
2                    2         Graduate degree  
4                    2         Graduate degree  
5                   30                   Other  
..                 ...                     ...  
489                  2         Graduate degree  
490                  0            Some college  
493                 10            Some college  
494                  3         Graduate degree  
495                  7             High school  

[338 rows x 12 columns]

In [ ]:
# Función para limpiar las columnas en fomrato de horas semanales
def clean_weekly_hours(value):
    try:
        # Elimina cualquier carácter no numérico y espacios en blanco
        value = value.strip().lower().replace('hrs', '').replace(' ', '')

        if '-' in value:  # Convierte intervalos en sus promedios
            start, end = map(float, value.split('-'))
            return round((start + end) / 2)
        elif value.isdigit() or value.replace('.', '', 1).isdigit():  # Maneja números y decimales
            value = float(value)
            if 0 <= value <= 168:  # Considera valores dentro de un rango realista para horas de sueño
                return int(round(value))
        return np.nan  # Asigna NaN a valores inválidos o atípicos
    except:
        return np.nan  # Maneja cualquier caso inesperado

# Aplica la misma función de limpieza a la columna 'Paid_Work_Hours' en los datos de Minesota
mn_df['Paid_Work_Hours'] = mn_df['Paid_Work_Hours'].apply(lambda x: clean_weekly_hours(str(x)))

# Elimina las filas con NaN en 'Paid_Work_Hours' (valores atípicos y entradas inválidas)
mn_df = mn_df.dropna(subset=['Paid_Work_Hours'])

# Convierte la columna 'Paid_Work_Hours' a tipo entero
mn_df['Paid_Work_Hours'] = mn_df['Paid_Work_Hours'].astype(int)

# Muestra el dataframe limpio
mn_df[['Paid_Work_Hours']]
#len(ne_df)
mn_df

<ipython-input-6-5e1e39adf03e>:25: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  mn_df['Paid_Work_Hours'] = mn_df['Paid_Work_Hours'].astype(int)


Region  Gender        Handed Favourite_physical_activity  \
0       MN  Female  Right-Handed                Hockey (Ice)   
1       MN    Male  Right-Handed              Hockey (Field)   
2       MN    Male  Right-Handed           Baseball/Softball   
4       MN  Female  Right-Handed                Martial Arts   
5       MN    Male  Right-Handed                       Other   
..     ...     ...           ...                         ...   
489     MN  Female  Right-Handed                     Dancing   
490     MN  Female  Right-Handed                    Swimming   
493     MN    Male  Right-Handed         Football (American)   
494     MN  Female  Right-Handed                    Swimming   
495     MN  Female  Right-Handed                      Tennis   

     Importance_reducing_pollution Birth_month                  Beverage  \
0                              316     October                     Water   
1                              690        July                     Water   
2                             1000     October                     Water   
4                              999    February                     Water   
5                              396     January            Powdered drink   
..                             ...         ...                       ...   
489                            800     October                       Tea   
490                            519    February                       Tea   
493                            675   September  Soft drink (caffeinated)   
494                            900         May                     Water   
495                            765        June                     Water   

    Favorite_School_Subject  Sleep_Hours_Non_Schoolnight  Paid_Work_Hours  \
0                   English                           10               20   
1                   Science                            8               15   
2        Physical education                           12                2   
4                       Art                            5               40   
5        Physical education                           14                0   
..                      ...                          ...              ...   
489                 Science                            9                0   
490                 Science                            4                0   
493      Physical education                           10               10   
494                 Science                           11               10   
495                   Music                            9               12   

    Work_At_Home_Hours Planned_Education_Level  
0                    0         Graduate degree  
1                    8   Less than high school  
2                    2         Graduate degree  
4                    2         Graduate degree  
5                   30                   Other  
..                 ...                     ...  
489                  2         Graduate degree  
490                  0            Some college  
493                 10            Some college  
494                  3         Graduate degree  
495                  7             High school  

[336 rows x 12 columns]

In [ ]:
# Aplica la misma función de limpieza a la columna 'Work_At_Home_Hours' en los datos de Minesota
mn_df['Work_At_Home_Hours'] = mn_df['Work_At_Home_Hours'].apply(lambda x: clean_weekly_hours(str(x)))

# Elimina las filas con NaN en 'Paid_Work_Hours' (valores atípicos y entradas inválidas)
mn_df = mn_df.dropna(subset=['Work_At_Home_Hours'])

# Convierte la columna 'Paid_Work_Hours' a tipo entero
mn_df['Work_At_Home_Hours'] = mn_df['Work_At_Home_Hours'].astype(int)

# Muestra el dataframe limpio
mn_df[['Work_At_Home_Hours']]
#len(ne_df)
mn_df

<ipython-input-7-af260d4a41eb>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  mn_df['Work_At_Home_Hours'] = mn_df['Work_At_Home_Hours'].apply(lambda x: clean_weekly_hours(str(x)))
<ipython-input-7-af260d4a41eb>:8: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  mn_df['Work_At_Home_Hours'] = mn_df['Work_At_Home_Hours'].astype(int)


Region  Gender        Handed Favourite_physical_activity  \
0       MN  Female  Right-Handed                Hockey (Ice)   
1       MN    Male  Right-Handed              Hockey (Field)   
2       MN    Male  Right-Handed           Baseball/Softball   
4       MN  Female  Right-Handed                Martial Arts   
5       MN    Male  Right-Handed                       Other   
..     ...     ...           ...                         ...   
489     MN  Female  Right-Handed                     Dancing   
490     MN  Female  Right-Handed                    Swimming   
493     MN    Male  Right-Handed         Football (American)   
494     MN  Female  Right-Handed                    Swimming   
495     MN  Female  Right-Handed                      Tennis   

     Importance_reducing_pollution Birth_month                  Beverage  \
0                              316     October                     Water   
1                              690        July                     Water   
2                             1000     October                     Water   
4                              999    February                     Water   
5                              396     January            Powdered drink   
..                             ...         ...                       ...   
489                            800     October                       Tea   
490                            519    February                       Tea   
493                            675   September  Soft drink (caffeinated)   
494                            900         May                     Water   
495                            765        June                     Water   

    Favorite_School_Subject  Sleep_Hours_Non_Schoolnight  Paid_Work_Hours  \
0                   English                           10               20   
1                   Science                            8               15   
2        Physical education                           12                2   
4                       Art                            5               40   
5        Physical education                           14                0   
..                      ...                          ...              ...   
489                 Science                            9                0   
490                 Science                            4                0   
493      Physical education                           10               10   
494                 Science                           11               10   
495                   Music                            9               12   

     Work_At_Home_Hours Planned_Education_Level  
0                     0         Graduate degree  
1                     8   Less than high school  
2                     2         Graduate degree  
4                     2         Graduate degree  
5                    30                   Other  
..                  ...                     ...  
489                   2         Graduate degree  
490                   0            Some college  
493                  10            Some college  
494                   3         Graduate degree  
495                   7             High school  

[334 rows x 12 columns]

# Nebraska's DataFrame


In [ ]:
# Se crea el dataframe de nebraska desde el CSV en el repositorio de github
ne_df = pd.read_csv("https://raw.githubusercontent.com/valentinabc19/Proyecto1-Estadistica1/main/samples/nebraska_sample.csv")

# Se seleccionan solo las columnas de interés
ne_df = ne_df[['Region','Gender','Handed', 'Favourite_physical_activity', 'Importance_reducing_pollution',
               'Birth_month', 'Beverage', 'Favorite_School_Subject', 'Sleep_Hours_Non_Schoolnight',
               'Paid_Work_Hours', 'Work_At_Home_Hours','Planned_Education_Level']]

# Mostrar el DataFrame
ne_df
len(ne_df) #Verificación de total de filas

500

In [ ]:
ne_df = ne_df.dropna() #Se eliminan las filas con valores nulos
len(ne_df) #Verificación de total de filas restantes
ne_df

Region  Gender        Handed Favourite_physical_activity  \
0       NE  Female  Right-Handed                  Basketball   
1       NE    Male  Ambidextrous                       Other   
2       NE  Female  Right-Handed                      Soccer   
4       NE  Female   Left-Handed                      Soccer   
5       NE    Male  Right-Handed                  Basketball   
..     ...     ...           ...                         ...   
492     NE  Female  Right-Handed             Running/Jogging   
493     NE    Male   Left-Handed                  Basketball   
494     NE    Male  Right-Handed         Football (American)   
495     NE    Male  Ambidextrous                       Other   
499     NE  Female  Right-Handed                  Basketball   

     Importance_reducing_pollution Birth_month                  Beverage  \
0                            950.0        June                     Water   
1                            647.0    February                       Tea   
2                            900.0       March                     Water   
4                            743.0         May                     Water   
5                           1000.0     October              Sports drink   
..                             ...         ...                       ...   
492                            1.0         May                     Water   
493                           50.0    December                     Water   
494                          878.0   September                     Water   
495                          800.0    December              Sports drink   
499                          700.0    November  Soft drink (caffeinated)   

        Favorite_School_Subject Sleep_Hours_Non_Schoolnight Paid_Work_Hours  \
0                       Science                           9               0   
1            Physical education                           9              20   
2                         Other                           8               3   
4                       History                          10               0   
5                       Science                           7               0   
..                          ...                         ...             ...   
492    Computers and technology                           9               0   
493                     History                          12               3   
494          Physical education                           8               3   
495                     History                           4               0   
499  Mathematics and statistics                          11              40   

    Work_At_Home_Hours Planned_Education_Level  
0                    5         Graduate degree  
1                   10   Less than high school  
2                    2         Graduate degree  
4                    5         Graduate degree  
5                    1            Some college  
..                 ...                     ...  
492                  3    Undergraduate degree  
493                  5         Graduate degree  
494                  0         Graduate degree  
495                  6             High school  
499                 60         Graduate degree  

[387 rows x 12 columns]

In [ ]:
# Convertir la columna 'Importance_reducing_pollution' a numérico y manejar los valores no numéricos de forma adecuada
ne_df['Importance_reducing_pollution'] = pd.to_numeric(ne_df['Importance_reducing_pollution'], errors='coerce').astype(int)

# Eliminar valores que no están dentro del rango válido (0 a 1000)
ne_df = ne_df[(ne_df['Importance_reducing_pollution'] >= 0) & (ne_df['Importance_reducing_pollution'] <= 1000)]

ne_df[['Importance_reducing_pollution']]
len(ne_df)

<ipython-input-10-118c31f7ed4c>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  ne_df['Importance_reducing_pollution'] = pd.to_numeric(ne_df['Importance_reducing_pollution'], errors='coerce').astype(int)


386

In [ ]:
# Aplica la misma función de limpieza a la columna 'Sleep_Hours_Non_Schoolnight' en los datos de Nebraska
ne_df['Sleep_Hours_Non_Schoolnight'] = ne_df['Sleep_Hours_Non_Schoolnight'].apply(lambda x: clean_sleep_hours(str(x)))

# Elimina las filas con NaN en 'Sleep_Hours_Non_Schoolnight' (valores atípicos y entradas inválidas)
ne_df = ne_df.dropna(subset=['Sleep_Hours_Non_Schoolnight'])

# Convierte la columna 'Sleep_Hours_Non_Schoolnight' a tipo entero
ne_df['Sleep_Hours_Non_Schoolnight'] = ne_df['Sleep_Hours_Non_Schoolnight'].astype(int)

# Muestra el dataframe limpio
ne_df[['Sleep_Hours_Non_Schoolnight']]
len(ne_df)
#ne_df

<ipython-input-11-f1b582968d39>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  ne_df['Sleep_Hours_Non_Schoolnight'] = ne_df['Sleep_Hours_Non_Schoolnight'].apply(lambda x: clean_sleep_hours(str(x)))


385

In [ ]:
# Aplica la misma función de limpieza a la columna 'Paid_Work_Hours' en los datos de Nebraska
ne_df['Paid_Work_Hours'] = ne_df['Paid_Work_Hours'].apply(lambda x: clean_weekly_hours(str(x)))

# Elimina las filas con NaN en 'Paid_Work_Hours' (valores atípicos y entradas inválidas)
ne_df = ne_df.dropna(subset=['Paid_Work_Hours'])

# Convierte la columna 'Paid_Work_Hours' a tipo entero
ne_df['Paid_Work_Hours'] = ne_df['Paid_Work_Hours'].astype(int)

# Muestra el dataframe limpio
ne_df[['Paid_Work_Hours']]
#len(ne_df)
ne_df

Region  Gender        Handed Favourite_physical_activity  \
0       NE  Female  Right-Handed                  Basketball   
1       NE    Male  Ambidextrous                       Other   
2       NE  Female  Right-Handed                      Soccer   
4       NE  Female   Left-Handed                      Soccer   
5       NE    Male  Right-Handed                  Basketball   
..     ...     ...           ...                         ...   
492     NE  Female  Right-Handed             Running/Jogging   
493     NE    Male   Left-Handed                  Basketball   
494     NE    Male  Right-Handed         Football (American)   
495     NE    Male  Ambidextrous                       Other   
499     NE  Female  Right-Handed                  Basketball   

     Importance_reducing_pollution Birth_month                  Beverage  \
0                              950        June                     Water   
1                              647    February                       Tea   
2                              900       March                     Water   
4                              743         May                     Water   
5                             1000     October              Sports drink   
..                             ...         ...                       ...   
492                              1         May                     Water   
493                             50    December                     Water   
494                            878   September                     Water   
495                            800    December              Sports drink   
499                            700    November  Soft drink (caffeinated)   

        Favorite_School_Subject  Sleep_Hours_Non_Schoolnight  Paid_Work_Hours  \
0                       Science                            9                0   
1            Physical education                            9               20   
2                         Other                            8                3   
4                       History                           10                0   
5                       Science                            7                0   
..                          ...                          ...              ...   
492    Computers and technology                            9                0   
493                     History                           12                3   
494          Physical education                            8                3   
495                     History                            4                0   
499  Mathematics and statistics                           11               40   

    Work_At_Home_Hours Planned_Education_Level  
0                    5         Graduate degree  
1                   10   Less than high school  
2                    2         Graduate degree  
4                    5         Graduate degree  
5                    1            Some college  
..                 ...                     ...  
492                  3    Undergraduate degree  
493                  5         Graduate degree  
494                  0         Graduate degree  
495                  6             High school  
499                 60         Graduate degree  

[385 rows x 12 columns]

In [ ]:
# Aplica la misma función de limpieza a la columna 'Work_At_Home_Hours' en los datos de Nebraska
ne_df['Work_At_Home_Hours'] = ne_df['Work_At_Home_Hours'].apply(lambda x: clean_weekly_hours(str(x)))

# Elimina las filas con NaN en 'Paid_Work_Hours' (valores atípicos y entradas inválidas)
ne_df = ne_df.dropna(subset=['Work_At_Home_Hours'])

# Convierte la columna 'Paid_Work_Hours' a tipo entero
ne_df['Work_At_Home_Hours'] = ne_df['Work_At_Home_Hours'].astype(int)

# Muestra el dataframe limpio
ne_df[['Work_At_Home_Hours']]
#len(ne_df)
ne_df

<ipython-input-13-4757090b6c4c>:8: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  ne_df['Work_At_Home_Hours'] = ne_df['Work_At_Home_Hours'].astype(int)


Region  Gender        Handed Favourite_physical_activity  \
0       NE  Female  Right-Handed                  Basketball   
1       NE    Male  Ambidextrous                       Other   
2       NE  Female  Right-Handed                      Soccer   
4       NE  Female   Left-Handed                      Soccer   
5       NE    Male  Right-Handed                  Basketball   
..     ...     ...           ...                         ...   
492     NE  Female  Right-Handed             Running/Jogging   
493     NE    Male   Left-Handed                  Basketball   
494     NE    Male  Right-Handed         Football (American)   
495     NE    Male  Ambidextrous                       Other   
499     NE  Female  Right-Handed                  Basketball   

     Importance_reducing_pollution Birth_month                  Beverage  \
0                              950        June                     Water   
1                              647    February                       Tea   
2                              900       March                     Water   
4                              743         May                     Water   
5                             1000     October              Sports drink   
..                             ...         ...                       ...   
492                              1         May                     Water   
493                             50    December                     Water   
494                            878   September                     Water   
495                            800    December              Sports drink   
499                            700    November  Soft drink (caffeinated)   

        Favorite_School_Subject  Sleep_Hours_Non_Schoolnight  Paid_Work_Hours  \
0                       Science                            9                0   
1            Physical education                            9               20   
2                         Other                            8                3   
4                       History                           10                0   
5                       Science                            7                0   
..                          ...                          ...              ...   
492    Computers and technology                            9                0   
493                     History                           12                3   
494          Physical education                            8                3   
495                     History                            4                0   
499  Mathematics and statistics                           11               40   

     Work_At_Home_Hours Planned_Education_Level  
0                     5         Graduate degree  
1                    10   Less than high school  
2                     2         Graduate degree  
4                     5         Graduate degree  
5                     1            Some college  
..                  ...                     ...  
492                   3    Undergraduate degree  
493                   5         Graduate degree  
494                   0         Graduate degree  
495                   6             High school  
499                  60         Graduate degree  

[382 rows x 12 columns]


# Rhode Island's DataFrame


In [ ]:
# Se crea el dataframe de rhode island desde el CSV en el repositorio de github
ri_df = pd.read_csv("https://raw.githubusercontent.com/valentinabc19/Proyecto1-Estadistica1/main/samples/rhode_island_sample.csv")

# Se seleccionan solo las columnas de interés
ri_df = ri_df[['Region','Gender','Handed', 'Favourite_physical_activity', 'Importance_reducing_pollution',
               'Birth_month', 'Beverage', 'Favorite_School_Subject', 'Sleep_Hours_Non_Schoolnight',
               'Paid_Work_Hours', 'Work_At_Home_Hours','Planned_Education_Level']]

# Mostrar el DataFrame
ri_df
len(ri_df) #Verificación de total de filas

500

In [ ]:
ri_df = ri_df.dropna() #Se eliminan las filas con valores nulos
len(ri_df) #Verificación de total de filas restantes
#ri_df

355

In [ ]:
# Convertir la columna 'Importance_reducing_pollution' a numérico y manejar los valores no numéricos de forma adecuada
ri_df['Importance_reducing_pollution'] = pd.to_numeric(ri_df['Importance_reducing_pollution'], errors='coerce').astype(int)

# Eliminar valores que no están dentro del rango válido (0 a 1000)
ri_df = ri_df[(ri_df['Importance_reducing_pollution'] >= 0) & (ri_df['Importance_reducing_pollution'] <= 1000)]

ri_df[['Importance_reducing_pollution']]

<ipython-input-16-56ea62d5108a>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  ri_df['Importance_reducing_pollution'] = pd.to_numeric(ri_df['Importance_reducing_pollution'], errors='coerce').astype(int)


Importance_reducing_pollution
0                              680
1                              872
2                              900
3                             1000
5                             1000
..                             ...
494                            100
495                            802
496                           1000
497                            695
499                            750

[353 rows x 1 columns]

In [ ]:
# Aplica la misma función de limpieza a la columna 'Sleep_Hours_Non_Schoolnight' en los datos de Rhode Island
ri_df['Sleep_Hours_Non_Schoolnight'] = ri_df['Sleep_Hours_Non_Schoolnight'].apply(lambda x: clean_sleep_hours(str(x)))

# Elimina las filas con NaN en 'Sleep_Hours_Non_Schoolnight' (valores atípicos y entradas inválidas)
ri_df = ri_df.dropna(subset=['Sleep_Hours_Non_Schoolnight'])

# Convierte la columna 'Sleep_Hours_Non_Schoolnight' a tipo entero
ri_df['Sleep_Hours_Non_Schoolnight'] = ri_df['Sleep_Hours_Non_Schoolnight'].astype(int)

# Muestra el dataframe limpio
ri_df[['Sleep_Hours_Non_Schoolnight']]
ri_df

Region  Gender        Handed Favourite_physical_activity  \
0       RI    Male  Right-Handed         Football (American)   
1       RI    Male  Right-Handed              Walking/Hiking   
2       RI    Male  Right-Handed                      Soccer   
3       RI  Female  Right-Handed                      Soccer   
5       RI    Male  Right-Handed                      Soccer   
..     ...     ...           ...                         ...   
494     RI  Female  Right-Handed              Walking/Hiking   
495     RI  Female  Right-Handed              Walking/Hiking   
496     RI    Male   Left-Handed           Baseball/Softball   
497     RI  Female  Right-Handed                      Tennis   
499     RI    Male  Right-Handed                  Basketball   

     Importance_reducing_pollution Birth_month                  Beverage  \
0                              680        June                     Water   
1                              872        July                     Water   
2                              900        June                     Water   
3                             1000    November                     Juice   
5                             1000     October  Soft drink (caffeinated)   
..                             ...         ...                       ...   
494                            100        June                     Water   
495                            802        June                     Water   
496                           1000      August              Sports drink   
497                            695       March  Soft drink (caffeinated)   
499                            750    December                     Water   

        Favorite_School_Subject  Sleep_Hours_Non_Schoolnight Paid_Work_Hours  \
0    Mathematics and statistics                           11               0   
1                       Science                           10               0   
2    Mathematics and statistics                            8               0   
3                         Other                            5              30   
5                       History                            9              20   
..                          ...                          ...             ...   
494                       Other                            8           30-40   
495                         Art                            9               1   
496              Social studies                            3               5   
497                         Art                            9               0   
499                     Science                            6               0   

    Work_At_Home_Hours Planned_Education_Level  
0                    4         Graduate degree  
1                    5         Graduate degree  
2                    5                   Other  
3                   30            Some college  
5                    3         Graduate degree  
..                 ...                     ...  
494                  8         Graduate degree  
495                  5         Graduate degree  
496                  7         Graduate degree  
497                  2    Undergraduate degree  
499                  1            Some college  

[352 rows x 12 columns]

In [ ]:
# Aplica la misma función de limpieza a la columna 'Paid_Work_Hours' en los datos de Rhode Island
ri_df['Paid_Work_Hours'] = ri_df['Paid_Work_Hours'].apply(lambda x: clean_weekly_hours(str(x)))

# Elimina las filas con NaN en 'Paid_Work_Hours' (valores atípicos y entradas inválidas)
ri_df = ri_df.dropna(subset=['Paid_Work_Hours'])

# Convierte la columna 'Paid_Work_Hours' a tipo entero
ri_df['Paid_Work_Hours'] = ri_df['Paid_Work_Hours'].astype(int)

# Muestra el dataframe limpio
ri_df[['Paid_Work_Hours']]
#len(ri_df)
ri_df

<ipython-input-18-e96323bbff46>:8: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  ri_df['Paid_Work_Hours'] = ri_df['Paid_Work_Hours'].astype(int)


Region  Gender        Handed Favourite_physical_activity  \
0       RI    Male  Right-Handed         Football (American)   
1       RI    Male  Right-Handed              Walking/Hiking   
2       RI    Male  Right-Handed                      Soccer   
3       RI  Female  Right-Handed                      Soccer   
5       RI    Male  Right-Handed                      Soccer   
..     ...     ...           ...                         ...   
494     RI  Female  Right-Handed              Walking/Hiking   
495     RI  Female  Right-Handed              Walking/Hiking   
496     RI    Male   Left-Handed           Baseball/Softball   
497     RI  Female  Right-Handed                      Tennis   
499     RI    Male  Right-Handed                  Basketball   

     Importance_reducing_pollution Birth_month                  Beverage  \
0                              680        June                     Water   
1                              872        July                     Water   
2                              900        June                     Water   
3                             1000    November                     Juice   
5                             1000     October  Soft drink (caffeinated)   
..                             ...         ...                       ...   
494                            100        June                     Water   
495                            802        June                     Water   
496                           1000      August              Sports drink   
497                            695       March  Soft drink (caffeinated)   
499                            750    December                     Water   

        Favorite_School_Subject  Sleep_Hours_Non_Schoolnight  Paid_Work_Hours  \
0    Mathematics and statistics                           11                0   
1                       Science                           10                0   
2    Mathematics and statistics                            8                0   
3                         Other                            5               30   
5                       History                            9               20   
..                          ...                          ...              ...   
494                       Other                            8               35   
495                         Art                            9                1   
496              Social studies                            3                5   
497                         Art                            9                0   
499                     Science                            6                0   

    Work_At_Home_Hours Planned_Education_Level  
0                    4         Graduate degree  
1                    5         Graduate degree  
2                    5                   Other  
3                   30            Some college  
5                    3         Graduate degree  
..                 ...                     ...  
494                  8         Graduate degree  
495                  5         Graduate degree  
496                  7         Graduate degree  
497                  2    Undergraduate degree  
499                  1            Some college  

[344 rows x 12 columns]

In [ ]:
# Aplica la misma función de limpieza a la columna 'Work_At_Home_Hours' en los datos de Rhode Island
ri_df['Work_At_Home_Hours'] = ri_df['Work_At_Home_Hours'].apply(lambda x: clean_weekly_hours(str(x)))

# Elimina las filas con NaN en 'Paid_Work_Hours' (valores atípicos y entradas inválidas)
ri_df = ri_df.dropna(subset=['Work_At_Home_Hours'])

# Convierte la columna 'Paid_Work_Hours' a tipo entero
ri_df['Work_At_Home_Hours'] = ri_df['Work_At_Home_Hours'].astype(int)

# Muestra el dataframe limpio
ri_df[['Work_At_Home_Hours']]
#len(ri_df)
ri_df

<ipython-input-19-5d3808bfe44f>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  ri_df['Work_At_Home_Hours'] = ri_df['Work_At_Home_Hours'].apply(lambda x: clean_weekly_hours(str(x)))
<ipython-input-19-5d3808bfe44f>:8: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  ri_df['Work_At_Home_Hours'] = ri_df['Work_At_Home_Hours'].astype(int)


Region  Gender        Handed Favourite_physical_activity  \
0       RI    Male  Right-Handed         Football (American)   
1       RI    Male  Right-Handed              Walking/Hiking   
2       RI    Male  Right-Handed                      Soccer   
3       RI  Female  Right-Handed                      Soccer   
5       RI    Male  Right-Handed                      Soccer   
..     ...     ...           ...                         ...   
494     RI  Female  Right-Handed              Walking/Hiking   
495     RI  Female  Right-Handed              Walking/Hiking   
496     RI    Male   Left-Handed           Baseball/Softball   
497     RI  Female  Right-Handed                      Tennis   
499     RI    Male  Right-Handed                  Basketball   

     Importance_reducing_pollution Birth_month                  Beverage  \
0                              680        June                     Water   
1                              872        July                     Water   
2                              900        June                     Water   
3                             1000    November                     Juice   
5                             1000     October  Soft drink (caffeinated)   
..                             ...         ...                       ...   
494                            100        June                     Water   
495                            802        June                     Water   
496                           1000      August              Sports drink   
497                            695       March  Soft drink (caffeinated)   
499                            750    December                     Water   

        Favorite_School_Subject  Sleep_Hours_Non_Schoolnight  Paid_Work_Hours  \
0    Mathematics and statistics                           11                0   
1                       Science                           10                0   
2    Mathematics and statistics                            8                0   
3                         Other                            5               30   
5                       History                            9               20   
..                          ...                          ...              ...   
494                       Other                            8               35   
495                         Art                            9                1   
496              Social studies                            3                5   
497                         Art                            9                0   
499                     Science                            6                0   

     Work_At_Home_Hours Planned_Education_Level  
0                     4         Graduate degree  
1                     5         Graduate degree  
2                     5                   Other  
3                    30            Some college  
5                     3         Graduate degree  
..                  ...                     ...  
494                   8         Graduate degree  
495                   5         Graduate degree  
496                   7         Graduate degree  
497                   2    Undergraduate degree  
499                   1            Some college  

[341 rows x 12 columns]

# Wisconsin's DataFrame

In [ ]:
# Se crea el dataframe de wisconsin desde el CSV en el repositorio de github
wi_df = pd.read_csv("https://raw.githubusercontent.com/valentinabc19/Proyecto1-Estadistica1/main/samples/wisconsin_sample.csv")

# Se seleccionan solo las columnas de interés
wi_df = wi_df[['Region','Gender','Handed', 'Favourite_physical_activity', 'Importance_reducing_pollution',
               'Birth_month', 'Beverage', 'Favorite_School_Subject', 'Sleep_Hours_Non_Schoolnight',
               'Paid_Work_Hours', 'Work_At_Home_Hours','Planned_Education_Level']]

# Mostrar el DataFrame
wi_df
len(wi_df) #Verificación de total de filas

500

In [ ]:
wi_df = wi_df.dropna() #Se eliminan las filas con valores nulos
len(wi_df) #Verificación de total de filas restantes
#wi_df

374

In [ ]:
# Convertir la columna 'Importance_reducing_pollution' a numérico y manejar los valores no numéricos de forma adecuada
wi_df['Importance_reducing_pollution'] = pd.to_numeric(wi_df['Importance_reducing_pollution'], errors='coerce').astype(int)

# Eliminar valores que no están dentro del rango válido (0 a 1000)
wi_df = wi_df[(wi_df['Importance_reducing_pollution'] >= 0) & (wi_df['Importance_reducing_pollution'] <= 1000)]

wi_df[['Importance_reducing_pollution']]
len(wi_df)

370

In [ ]:
# Aplica la misma función de limpieza a la columna 'Sleep_Hours_Non_Schoolnight' en los datos de Wisconsin
wi_df['Sleep_Hours_Non_Schoolnight'] = wi_df['Sleep_Hours_Non_Schoolnight'].apply(lambda x: clean_sleep_hours(str(x)))

# Elimina las filas con NaN en 'Sleep_Hours_Non_Schoolnight' (valores atípicos y entradas inválidas)
wi_df = wi_df.dropna(subset=['Sleep_Hours_Non_Schoolnight'])

# Convierte la columna 'Sleep_Hours_Non_Schoolnight' a tipo entero
wi_df['Sleep_Hours_Non_Schoolnight'] = wi_df['Sleep_Hours_Non_Schoolnight'].astype(int)

# Muestra el dataframe limpio
wi_df[['Sleep_Hours_Non_Schoolnight']]
#wi_df
len(wi_df)

370

In [ ]:
# Aplica la misma función de limpieza a la columna 'Paid_Work_Hours' en los datos de Wisconsin
wi_df['Paid_Work_Hours'] = wi_df['Paid_Work_Hours'].apply(lambda x: clean_weekly_hours(str(x)))

# Elimina las filas con NaN en 'Paid_Work_Hours' (valores atípicos y entradas inválidas)
wi_df = wi_df.dropna(subset=['Paid_Work_Hours'])

# Convierte la columna 'Paid_Work_Hours' a tipo entero
wi_df['Paid_Work_Hours'] = wi_df['Paid_Work_Hours'].astype(int)

# Muestra el dataframe limpio
wi_df[['Paid_Work_Hours']]
#len(wi_df)
wi_df

<ipython-input-24-d3c2d4ed4e5a>:8: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  wi_df['Paid_Work_Hours'] = wi_df['Paid_Work_Hours'].astype(int)


Region  Gender        Handed Favourite_physical_activity  \
0       WI    Male  Right-Handed           Baseball/Softball   
2       WI  Female  Right-Handed                     Dancing   
4       WI  Female  Right-Handed                       Other   
5       WI  Female  Right-Handed                    Swimming   
8       WI  Female  Right-Handed                Hockey (Ice)   
..     ...     ...           ...                         ...   
494     WI    Male  Right-Handed                      Soccer   
495     WI  Female  Right-Handed                      Soccer   
496     WI  Female  Right-Handed                      Soccer   
498     WI  Female  Right-Handed                      Soccer   
499     WI    Male  Right-Handed         Football (American)   

     Importance_reducing_pollution Birth_month Beverage  \
0                              882       April    Water   
2                             1000   September   Coffee   
4                              200         May    Water   
5                             1000   September    Water   
8                              900   September    Water   
..                             ...         ...      ...   
494                              0      August    Water   
495                            684        June    Water   
496                            700       March    Water   
498                            652         May    Water   
499                           1000     October    Water   

    Favorite_School_Subject  Sleep_Hours_Non_Schoolnight  Paid_Work_Hours  \
0            Social studies                            9                0   
2                   Science                            8               15   
4                   Science                            9                0   
5                   History                           11                0   
8                   Science                           10                0   
..                      ...                          ...              ...   
494      Physical education                           10                0   
495                 Science                           11                0   
496                     Art                            9               10   
498                 Science                            5                0   
499      Physical education                            9                0   

    Work_At_Home_Hours Planned_Education_Level  
0                    1         Graduate degree  
2                    2         Graduate degree  
4                    2         Graduate degree  
5                    2         Graduate degree  
8                    0         Graduate degree  
..                 ...                     ...  
494                  2         Graduate degree  
495                  0         Graduate degree  
496                  2    Undergraduate degree  
498                  1    Undergraduate degree  
499                  0         Graduate degree  

[369 rows x 12 columns]

In [ ]:
# Aplica la misma función de limpieza a la columna 'Work_At_Home_Hours' en los datos de Wisconsin
wi_df['Work_At_Home_Hours'] = wi_df['Work_At_Home_Hours'].apply(lambda x: clean_weekly_hours(str(x)))

# Elimina las filas con NaN en 'Paid_Work_Hours' (valores atípicos y entradas inválidas)
wi_df = wi_df.dropna(subset=['Work_At_Home_Hours'])

# Convierte la columna 'Paid_Work_Hours' a tipo entero
wi_df['Work_At_Home_Hours'] = wi_df['Work_At_Home_Hours'].astype(int)

# Muestra el dataframe limpio
wi_df[['Work_At_Home_Hours']]
#len(wi_df)
wi_df

<ipython-input-25-19ee060bca61>:8: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  wi_df['Work_At_Home_Hours'] = wi_df['Work_At_Home_Hours'].astype(int)


Region  Gender        Handed Favourite_physical_activity  \
0       WI    Male  Right-Handed           Baseball/Softball   
2       WI  Female  Right-Handed                     Dancing   
4       WI  Female  Right-Handed                       Other   
5       WI  Female  Right-Handed                    Swimming   
8       WI  Female  Right-Handed                Hockey (Ice)   
..     ...     ...           ...                         ...   
494     WI    Male  Right-Handed                      Soccer   
495     WI  Female  Right-Handed                      Soccer   
496     WI  Female  Right-Handed                      Soccer   
498     WI  Female  Right-Handed                      Soccer   
499     WI    Male  Right-Handed         Football (American)   

     Importance_reducing_pollution Birth_month Beverage  \
0                              882       April    Water   
2                             1000   September   Coffee   
4                              200         May    Water   
5                             1000   September    Water   
8                              900   September    Water   
..                             ...         ...      ...   
494                              0      August    Water   
495                            684        June    Water   
496                            700       March    Water   
498                            652         May    Water   
499                           1000     October    Water   

    Favorite_School_Subject  Sleep_Hours_Non_Schoolnight  Paid_Work_Hours  \
0            Social studies                            9                0   
2                   Science                            8               15   
4                   Science                            9                0   
5                   History                           11                0   
8                   Science                           10                0   
..                      ...                          ...              ...   
494      Physical education                           10                0   
495                 Science                           11                0   
496                     Art                            9               10   
498                 Science                            5                0   
499      Physical education                            9                0   

     Work_At_Home_Hours Planned_Education_Level  
0                     1         Graduate degree  
2                     2         Graduate degree  
4                     2         Graduate degree  
5                     2         Graduate degree  
8                     0         Graduate degree  
..                  ...                     ...  
494                   2         Graduate degree  
495                   0         Graduate degree  
496                   2    Undergraduate degree  
498                   1    Undergraduate degree  
499                   0         Graduate degree  

[368 rows x 12 columns]

# General Dataframe

In [ ]:
df = pd.concat([mn_df, ne_df, ri_df, wi_df], ignore_index=True)
len(df)
df

Region  Gender        Handed Favourite_physical_activity  \
0        MN  Female  Right-Handed                Hockey (Ice)   
1        MN    Male  Right-Handed              Hockey (Field)   
2        MN    Male  Right-Handed           Baseball/Softball   
3        MN  Female  Right-Handed                Martial Arts   
4        MN    Male  Right-Handed                       Other   
...     ...     ...           ...                         ...   
1420     WI    Male  Right-Handed                      Soccer   
1421     WI  Female  Right-Handed                      Soccer   
1422     WI  Female  Right-Handed                      Soccer   
1423     WI  Female  Right-Handed                      Soccer   
1424     WI    Male  Right-Handed         Football (American)   

      Importance_reducing_pollution Birth_month        Beverage  \
0                               316     October           Water   
1                               690        July           Water   
2                              1000     October           Water   
3                               999    February           Water   
4                               396     January  Powdered drink   
...                             ...         ...             ...   
1420                              0      August           Water   
1421                            684        June           Water   
1422                            700       March           Water   
1423                            652         May           Water   
1424                           1000     October           Water   

     Favorite_School_Subject  Sleep_Hours_Non_Schoolnight  Paid_Work_Hours  \
0                    English                           10               20   
1                    Science                            8               15   
2         Physical education                           12                2   
3                        Art                            5               40   
4         Physical education                           14                0   
...                      ...                          ...              ...   
1420      Physical education                           10                0   
1421                 Science                           11                0   
1422                     Art                            9               10   
1423                 Science                            5                0   
1424      Physical education                            9                0   

      Work_At_Home_Hours Planned_Education_Level  
0                      0         Graduate degree  
1                      8   Less than high school  
2                      2         Graduate degree  
3                      2         Graduate degree  
4                     30                   Other  
...                  ...                     ...  
1420                   2         Graduate degree  
1421                   0         Graduate degree  
1422                   2    Undergraduate degree  
1423                   1    Undergraduate degree  
1424                   0         Graduate degree  

[1425 rows x 12 columns]

In [ ]:
'''
Este código es para descargar la nueva base de datos como CSV en el caso de que se quiera descargar este nuevo df unido y limpiado

import pandas as pd
from google.colab import files

# Guardar el DataFrame como un archivo CSV
df.to_csv('data_combined.csv', index=False)

# Descargar el archivo CSV
files.download('data_combined.csv')

'''

"\nEste código es para descargar la nueva base de datos como CSV en el caso de que se quiera descargar este nuevo df unido y limpiado\n\nimport pandas as pd\nfrom google.colab import files\n\n# Guardar el DataFrame como un archivo CSV\ndf.to_csv('data_combined.csv', index=False)\n\n# Descargar el archivo CSV\nfiles.download('data_combined.csv')\n\n"

# Preguntas y respuestas

##1. ¿Cuántos estudiantes son diestros, zurdos o ambidiestros?

In [ ]:
# Contar el número de estudiantes por tipo de "Handed"
handed_counts = df['Handed'].value_counts()

# Mostrar el resultado
print(handed_counts)

Handed
Right-Handed    1231
Left-Handed      116
Ambidextrous      78
Name: count, dtype: int64


##  2. ¿Cuál es el mes donde nacieron más estudiantes de los que participan en el censo escolar en cada estado?

In [ ]:
mes = df['Birth_month'].value_counts().idxmax()
conteo = df['Birth_month'].value_counts().max()
print(f"El mes en el que nacieron mas estudiantes es {mes} Con {conteo} registros")

El mes en el que nacieron mas estudiantes es March Con 143 registros


## 3. ¿Cuál es la bebida favorita de los estudiantes que participan en el Censo escolar en cada estado?

In [ ]:
df.groupby('Region')['Beverage'].agg(pd.Series.mode)

Region
MN    Water
NE    Water
RI    Water
WI    Water
Name: Beverage, dtype: object

## 4. ¿Cuál es el deporte / actividad favorita de los estudiantes que participan en el Censo escolar en cada estado?

In [ ]:
# Encuentra todas las actividades favoritas con el conteo máximo en cada estado
def get_favorite_activity(group):
    counts = group.value_counts()
    max_count = counts.max()
    favorites = counts[counts == max_count].index.tolist()
    return favorites

favorite_activities_by_state = df.groupby('Region')['Favourite_physical_activity'].agg(get_favorite_activity)

# Mostrar el resultado
print(favorite_activities_by_state)

Region
MN                [Other]
NE    [Basketball, Other]
RI               [Soccer]
WI                [Other]
Name: Favourite_physical_activity, dtype: object


## 5. ¿Cuál es el nivel más alto de educación que planean alcanzar los estudiantes que participan en el Censo escolar en cada estado?

In [ ]:
result = df.groupby('Region')['Planned_Education_Level'].agg(pd.Series.mode)

print(result)

Region
MN    Graduate degree
NE    Graduate degree
RI    Graduate degree
WI    Graduate degree
Name: Planned_Education_Level, dtype: object


## 6. Construya un histograma con la variable “tiempo que trabaja con pago el estudiante en la semana (medido en horas)” y a partir de  esta gráfica responda las siguientes preguntas:

A pesar de que en la gráfica se vea que el primer intervalo comienza en 0, el 0 no se esta teniendo en cuenta a la hora de graficar los datos.

In [ ]:
import plotly.express as px

# Filtrar el DataFrame para excluir las filas donde 'Paid_Work_Hours' es 0
filtered_df = df[df['Paid_Work_Hours'] != 0]  # Aquí usamos '> 0' para excluir 0

# Crear el histograma con el DataFrame filtrado
fig = px.histogram(
    filtered_df,  # Usar el DataFrame filtrado
    x='Paid_Work_Hours',
    nbins=16,
    title='Histograma de horas semanales de trabajo con pago',
    color_discrete_sequence=['#636EFA'],  # Cambiar el color
    opacity=0.75  # Hacer las barras un poco transparentes
)

fig.update_traces(marker=dict(line=dict(width=1, color='white')))

# Personalizar el layout del gráfico
fig.update_layout(
    xaxis_title='Horas semanales de trabajo con pago',
    yaxis_title='Frecuencia',
    xaxis=dict(range=[1, filtered_df['Paid_Work_Hours'].max()])  # Ajustar el rango de los bins
)

# Mostrar el histograma
fig.show()

### a) ¿Qué porcentaje de los estudiantes de la muestra trabajan máximo 13 horas a la semana con pago?

El total de los datos en la muestra sin contar los estudiantes que no trabajan es de 657.

Teniendo en cuenta los intervalos del histograma (0,19] debemos hacer una regla de 3 para hallar la cantidad de estudiantes que corresponden al intervalo de (0,13], el resultado de esta operación es 335. Considerando esto al hacer la operación para hallar el porcentaje, el resultado es 50.9%

### b) ¿Qué porcentaje de los estudiantes de la muestra trabajan mínimo 3 horas a la semana con pago?

Para calcular este porcentaje, necesitamos sumar las frecuencias de los estudiantes que trabajan al menos 3 horas por semana.

**Intervalos relevantes:**

- **Intervalo de 0 a 9 horas:** Hay 305 estudiantes. De estos, 6/9 (los que trabajan al menos 3 horas) representa la parte que estamos buscando.

Estudiantes que trabajan al menos 3 horas en el intervalo de 0 a 9 horas = 6/9 × 305 ≈ 203.33.

Luego, sumamos las frecuencias de los intervalos que están completamente por encima de 3 horas:

- **Intervalo de 10 a 19 horas:** 186 estudiantes
- **Intervalo de 20 a 29 horas:** 111 estudiantes
- **Intervalo de 30 a 39 horas:** 28 estudiantes
- **Intervalo de 40 a 49 horas:** 17 estudiantes
- **Intervalo de 50 a 59 horas:** 5 estudiantes
- **Intervalo de 60 a 69 horas:** 1 estudiante
- **Intervalo de 70 a 79 horas:** 2 estudiantes
- **Intervalo de 80 a 89 horas:** 1 estudiante

**Total de estudiantes que trabajan al menos 3 horas:**

203.33 + 186 + 111 + 28 + 17 + 5 + 1 + 2 + 1 = 554.33 ≈ 554

Por lo tanto, el porcentaje es:

Porcentaje = ( 554 / 656 )×100 ≈ 84.45%

**Aproximadamente el 84.5% de los estudiantes de la muestra trabajan al menos 3 horas a la semana con pago.**

### c) ¿El 20% de los estudiantes trabajan mínimo cuantas horas con pago?

El total de los datos en la muestra sin contar los estudiantes que no trabajan es de 657 registros.

Teniendo en cuenta los intervalos del histograma (0,19] primero hallamos la cantidad de registros que corresponden al 20% de los estudiantes que trabajan, que en este caso es de aproximadamente 131 estudiantes. Luego en el histograma observamos que esta cantidad de registros se encuentran en el primer intervalo (0,9] lo que significa que el 20% de los estudiantes trabajan minimo 1 hora con pago.

### d) ¿El 20% de los estudiantes trabajan máximo cuantas horas con pago?

Primero se debe hallar el 20% de los estudiantes, que son 131. Despues se debe aplicar la regla de 3 teniendo en cuenta que en el intervalo de (0,9] hay 305 estudiantes, así que se debe hallar el intervalo al que pertenecen los 131 estudiantes del 20%. El resultado de esta operación es es 3.8 que aproximandolo sería 4, que corresponde al intervalo (0,4]; por lo que el máximo de horas con pago que trabajan los estudiantes es 4.

### e) ¿En cuál intervalo se halla la moda?

La moda se encuentra en el intervalo 0 a 9 horas, ya que es el que tiene la mayor frecuencia con 305 estudiantes.

### f) ¿El 50% de los estudiantes trabajan máximo cuantas horas con pago?

El total de los datos en la muestra sin contar los estudiantes que no trabajan es de 657 registros.

Teniendo en cuenta los intervalos del histograma (0,19] primero hallamos la cantidad de registros que corresponden al 50% de los estudiantes que trabajan, que en este caso es de aproximadamente 328 estudiantes. Luego en el histograma observamos que esta cantidad de registros se encuentran entre el primer y segundo intervalo (0,19] por lo que es necesario calcular la cantidad aproximada de horas con pago que trabajan los estudiantes. En este caso haciendo uso de la regla de 3 se encontro que la cantidad maxima de horas que trabajan los estudiantes es de aproximadamente 13 horas.

### g) ¿En promedio cuantas horas trabajan con pago a la semana los estudiantes? ¿La desviación estándar?

In [ ]:
# Se halla el promedio teniendo en cuenta los datos de las horas de trabajo con pago de "filtered_df", el cual es el que se uso para generar el histograma.
mean_hours = filtered_df['Paid_Work_Hours'].mean()

#  Se halla la desviación estandar teniendo en cuenta los datos de las horas de trabajo con pago de "filtered_df", el cual es el que se uso para generar el histograma.
std_dev_hours = filtered_df['Paid_Work_Hours'].std()

print(f"Promedio de horas trabajadas con pago: {mean_hours:.2f}")
print(f"Desviación estándar de horas trabajadas con pago: {std_dev_hours:.2f}")


Promedio de horas trabajadas con pago: 12.54
Desviación estándar de horas trabajadas con pago: 11.59


### h) ¿Qué tipo de asimetría presenta esta variable?

Observando el gráfico, la mayoría de los datos se acumulan en los primeros intervalos (especialmente en el de 0 a 1 horas), y luego disminuyen drásticamente, con una larga cola hacia la derecha.

Este tipo de distribución es **asimetría positiva (sesgo a la derecha)**, porque la cola de la distribución se extiende más hacia la derecha que hacia la izquierda.

Así que, **la variable presenta asimetría positiva**.

## 7. Construya una Ojiva para la variable “Tiempo que dedican los estudiantes para ayudar a los quehaceres del Hogar” y a partir de esta gráfica responda las siguientes preguntas:

In [ ]:
import plotly.graph_objects as go

# Ordenar los datos por la columna de interés
df_sorted = df.sort_values(by='Work_At_Home_Hours')

# Definir el número de bins
bins = 20

# Calcular los bins manualmente y obtener los límites
df_sorted['Bins'] = pd.cut(df_sorted['Work_At_Home_Hours'], bins=bins)
bins_limits = pd.cut(df_sorted['Work_At_Home_Hours'], bins=bins, retbins=True)[1]

# Calcular la frecuencia acumulada por cada bin y convertirla a enteros
frecuencia_acumulada = df_sorted['Bins'].value_counts().sort_index().cumsum().astype(int)

# Obtener el punto medio de cada bin
bins_medio = 0.5 * (bins_limits[1:] + bins_limits[:-1])

# Crear la ojiva usando Plotly
fig = go.Figure()

fig.add_trace(go.Scatter(
    x=bins_medio,
    y=frecuencia_acumulada,
    mode='lines+markers',
    name='Frecuencia Acumulada'
))

fig.update_layout(
    title='Ojiva de "Tiempo dedicado a los quehaceres del Hogar"',
    xaxis_title='Horas dedicadas',
    yaxis_title='Frecuencia acumulada (número de estudiantes)',
    showlegend=True
)

fig.show()

### a) ¿Qué porcentaje de los estudiantes de la muestra dedican mínimo 15 horas a la semana para ayudar en sus casas?

El total de los datos en la muestra es 1425.

Teniendo en cuenta el punto de la ojiva (17.5,1383) que indica que 1383 de los estudiantes estan en 17.5 horas, para hallar el porcentaje de estudiantes que dedican minimo 15 horas a la semana a ayudar en sus casas, se debe hacer una regla de 3, (con el dato dado anteriormente) para hallar la cantidad de estudiantes que dedican minimo 15 horas a las tareas del hogar. El resultado de esto es  de 240 estudiantes, despúes de hacer la operación del porcentaje el resultado es de aproximadamente 16.84%. **Por lo que el porcentaje de los estudiantes que dedican minimo 15 horas a la semana para ayudar en sus casas es de aproximadamente 17%.**

### b) ¿Qué porcentaje de los estudiantes de la muestra dedican máximo 5 horas a la semana para ayudar en sus casas?

El total de los datos en la muestra es 1425.

Teniendo en cuenta el punto de la ojiva (7.5,1319) que indica que 1319 de lo estudiantes estan en 7.5 horas, para hallar el porcentaje de estudiantes que dedican maximo 5 horas a la semana a ayudar en sus casas, se debe hacer una regla de 3 (con el dato dado anteriormente) para hallar la cantidad de estudiantes que dedican máximo 5 horas a las tareas del hogas. El resultado de esto es 879, despúes de hacer la operación del porcentaje el resultado 61.68%. **Por lo que el porcentaje de los estudiantes que dedican máximo 5 horas a la semana para ayudar en sus casas.**

### c) ¿Cuál es el máximo número de horas que el 15% de los estudiantes ayudan en sus casas?

Para encontrar el número máximo de horas que el 15% de los estudiantes dedican a los quehaceres del hogar:

1. **15% de 1425 estudiantes** es aproximadamente 0.15 × 1425 = 213.75 estudiantes. Redondeamos a 214 estudiantes.
2. En la ojiva, buscamos el punto en el eje y donde la frecuencia acumulada es aproximadamente 214.

Tenemos que **1144 estudiantes dedican como máximo 2.45 horas**.

Queremos saber cuántas horas dedican **214 estudiantes**.

Para calcular el número máximo de horas que el 15% de los estudiantes dedican a los quehaceres del hogar utilizando una regla de tres, hacemos lo siguiente:

Horas = (2.45 * 214) / 1144

= 0.46

Entonces, el máximo número de horas que el 15% de los estudiantes dedican a los quehaceres del hogar es aproximadamente **0.46 horas** (que puede aproximarse a menos de 1 hora).

### d) ¿Qué porcentaje de los estudiantes ayudan en sus casas al menos 5 horas a la semana?

El total de los datos en la muestra es 1425.

Teniendo en cuenta el punto de la ojiva (7.5,1319) que indica que 1319 de los estudiantes estan en 7.5 horas, para hallar el porcentaje de estudiantes que dedican al menos 5 horas a la semana a ayudar en sus casas, se debe hacer una regla de 3, (con el dato dado anteriormente) para hallar la cantidad de estudiantes que dedican al menos 5 horas a las tareas del hogar. El resultado de esto es  de 546 estudiantes, despúes de hacer la operación del porcentaje el resultado es de aproximadamente 38.31%. **Por lo que el porcentaje de los estudiantes que dedican al menos 5 horas a la semana para ayudar en sus casas es de aproximadamente 38%.**

### e) ¿Cuál es el mínimo número de horas que el 35% de los estudiantes ayudan en sus casas?

El total de los datos en la muestra es de 1425.

Teniendo en cuenta el punto (2.45,1144) primero se halla la cantidad de registros que corresponden al 35% de los estudiantes, que en este caso es de aproximadamente 498 estudiantes. Luego, por una regla de 3, se halla que el intervalo es [0,1]. **Por lo que el mínimo de horas que el 35% de los estudiantes ayudan en sus casas es 0.**

###  f) ¿En cuál intervalo se halla la moda?

La moda es el intervalo con el mayor incremento en la frecuencia acumulada.

Observando la ojiva:

- El intervalo más empinado parece ser **entre 0 y 10 horas** (desde el primer punto (2.45, 1144) al segundo punto (7.5, 1319)).

Por lo tanto, el intervalo donde se halla la moda es **entre 0 y 10 horas**.

### g) ¿En promedio cuántas horas ayudan en sus casas en la semana los estudiantes? ¿Cuál es la desviación estándar?

In [ ]:
promedio_Home_Work = df['Work_At_Home_Hours'].mean()
desviacion_estandar_Home_Work = df['Work_At_Home_Hours'].std()
print(f"El promedio de horas de trabajo en casa de los estudiantes es: {promedio_Home_Work}")
print(f"La desviación estándar de horas de trabajo en casa de los estudiantes es: {desviacion_estandar_Home_Work}")

El promedio de horas de trabajo en casa de los estudiantes es: 4.754385964912281
La desviación estándar de horas de trabajo en casa de los estudiantes es: 8.754133657931348


###  h) ¿Qué tipo de asimetría presenta esta variable?

Observando el gráfico, la mayoría de los datos se acumulan en los primeros intervalos, y luego disminuyen drásticamente, con una larga cola hacia la derecha.

Este tipo de distribución es asimetría positiva (sesgo a la derecha), porque la cola de la distribución se extiende más hacia la derecha que hacia la izquierda.

Así que, la variable presenta asimetría positiva.

## 8. Construya una gráfica de Diagrama de cajas para la comparar la opinión en los diferentes Estados de hombres y mujeres (Gender) con respecto a la importancia de reducir la contaminación (pollution) a partir de esta grafica responda las siguientes preguntas:

In [ ]:
fig = px.box(
    df,
    x='Region',  # Variable categórica para el eje X
    y='Importance_reducing_pollution',            # Variable numérica a analizar
    color='Gender', # Variable categórica para desglosar
    title='Boxplot con Tres Variables',
    labels={
        'Región': 'Estado',
        'Importance_reducing_pollution': 'Importancia a reducir la polucion',
        'Gender': 'Sexo'
    }
)

# Mostrar el gráfico
fig.show()

### a) ¿En términos generales para quienes (hombres o mujeres) es más importante reducir la contaminación?

En términos generales, parece que es más importante para las mujeres reducir la contaminación. Esto se observa porque las cajas azules (representando a las mujeres) tienden a tener una mediana más alta que las cajas rojas (representando a los hombres) en todos los estados, lo que indica que en promedio las mujeres califican la importancia de reducir la contaminación más alta que los hombres.

### b) ¿En términos generales en cuál Estado es más importante reducir la contaminación?

En Rhode Island, ya que el 50% de los registros estan acumulados entre 1000 (Mucha importancia) y 508 (importancia media) para los hombres y entre 1000 y 588 para las mujeres.

### c) ¿En términos generales en cuál Estado es más importante para las mujeres reducir la contaminación?

En términos generales, en Rhode Island es más importante para las mujeres reducir la contaminación, ya que la mediana (línea en el medio de la caja) del boxplot para las mujeres en Rhode Island es la más alta en comparación con los otros estados.

### d) ¿En términos generales en cuál Estado es más importante para los hombres reducir la contaminación?

En términos generales, para los hombres, Rhode Island (MN) parece ser el estado donde es más importante reducir la contaminación. Esto se puede deducir porque la mediana de la caja roja (que representa a los hombres) en Rhode Island es más alta en comparación con los otros estados (NE, MN, WI).

### e) ¿En términos generales en qué género la importancia de reducir la contaminación es más homogénea?

En las mujeres la importancia de reducir la contaminacion es mas homgenea puesto que las cajas se ubican mas o menos a las mismas alturas y existe menos variabilidad en comparacion con las cajas de los hombres en donde las cajas tienen mas variabilidad.

### f) ¿En términos generales en cuál Estado la importancia de reducir la contaminación es más heterogénea?

En términos generales, la importancia de reducir la contaminación es más heterogénea en Nebraska. Esto se debe a que la caja del boxplot para Nebraska es la más larga, lo que indica una mayor dispersión de los datos y, por lo tanto, una mayor heterogeneidad en las opiniones sobre la importancia de reducir la contaminación en ese estado.

### g) ¿En cuál género se presenta mayor asimetría en las calificaciones de la importancia de reducir la contaminación?

El género masculino (representado en rojo) presenta mayor asimetría en las calificaciones de la importancia de reducir la contaminación, ya que las cajas rojas muestran una mayor variabilidad en la forma y rango de los bigotes en varias regiones, indicando una mayor dispersión y posible asimetría en comparación con las cajas azules que representan a las mujeres.

### h) ¿En cuál Estado se presenta mayor asimetría en las calificaciones de la importancia de reducir la contaminación?

A pesar de que en los demas estados existe una asimetria con respecto a las calificaciones de la importancia de reducir la contaminación, en Rhode Island se encuentra la mayor asimetria en esta variable porque la mayoria de los datos se encuentran acumulados a hacia arriba (si se hace una grafica mostrando la curva de distribucion de los datos en la asimetria los datos estarian acumulados a la derecha), ademas, no hay un bigote superior lo que signnifica que el 50% de los datos se acumula por debajo de primer cuartil.

### i) ¿En cuál Estado las mujeres presentan mayor simetría en las calificaciones de la importancia de reducir la contaminación?

Para determinar la simetría se debe observar la posición de la mediana resppecto a los bordes de la caja, que serían los cuartiles; entre más centrada este la línea de la mediana hay mayor simetría. Teniendo en cuenta esto, se puede concluir que las mujeres del estado de Winsconsin tienen mayor simetría, ya que la linea de la mediana parece estar casi en el centro de la caja.

## 9. Construya una tabla cruzada de indicadores entre la variable “¿Cuántas horas de sueño por noche suele dormir cuando no tiene clases al día siguiente?” versus “Género”. A partir de esta tabla responda las siguientes preguntas:

In [ ]:
crosstab_Sleep_Hours = pd.crosstab(df.Sleep_Hours_Non_Schoolnight, df.Gender)
crosstab_Sleep_Hours

Gender                       Female  Male
Sleep_Hours_Non_Schoolnight              
0                                 0     3
1                                 0     3
2                                 2     9
3                                 3    19
4                                13    25
5                                26    28
6                                44    53
7                                58    52
8                               127   125
9                               168   134
10                              169   133
11                               43    47
12                               49    52
13                                8     7
14                                6     8
15                                1     1
16                                2     1
18                                1     0
19                                1     1
20                                1     0
24                                2     0

### a) ¿Qué porcentaje de estudiantes de la muestra son mujeres?

Para calcular el porcentaje de estudiantes que son mujeres:

Porcentaje de mujeres = (Total de mujeres / Total de estudiantes) × 100

Sustituyendo los valores:

Porcentaje de mujeres = (724 / 1425) × 100 ≈ 50.81%

Entonces, aproximadamente **50.81%** de los estudiantes de la muestra son mujeres.

### b) ¿Cuántas horas de sueño en promedio suelen dormir cuando no tiene clases al día siguiente los estudiantes de la muestra?

Los estudiantes en promedio suelen dormir 9 horas aproximadamente cuando no tienen clase al dia siguiente. Se observa que las horas mas frecuentes son 8, 9 y 10.

In [ ]:
promedio = df['Sleep_Hours_Non_Schoolnight'].mean()
print(f"El promedio de horas de sueño en los estudiantes es: {promedio}")

El promedio de horas de sueño en los estudiantes es: 8.711578947368421


### c) ¿En términos generales que género duermen más tiempo?

El género que duerme más tiempo es el género "Femenino", ya que son las que tienen valores mayores en las horas más grandes.

###  d) ¿En términos generales qué género los tiempos son más homogéneos?

La homogeneidad en los tiempos de sueño puede interpretarse observando la concentración de datos en torno a ciertos valores. En la tabla:

- **Mujeres**: Los datos están más distribuidos entre diferentes horas de sueño, especialmente en el rango de 8 a 10 horas, pero con algunas frecuencias notables en horas como 9 y 10.
- **Hombres**: La distribución también es amplia, pero hay una mayor concentración de valores entre las 7 y 10 horas.

Para determinar la homogeneidad, analizamos la variabilidad:

- **Mujeres** muestran una distribución más uniforme en algunas horas, pero tienen picos más altos en horas específicas (9 y 10).
- **Hombres** tienen una concentración más alta en un rango estrecho (7 a 10 horas), pero menos distribución uniforme.

Dado que **los hombres** tienen más datos concentrados en menos categorías, podemos concluir que **los tiempos de sueño de los hombres son más homogéneos**.

###  e) ¿En términos generales qué género los tiempos son más simétricos?

Aunque no hay una gran diferencia entre las mujeres y los hombres, los tiempos son mas simetricos para los hombres puesto que los datos estan dispersos en una distancia menos variable con respecto a las mujeres.

### f) ¿Cuál género tiene un histograma más puntiagudo? ¿Cuál género tiene un histograma más aplanado o chato?

Los valores más altos se concentran en las horas de sueño de 8 a 10 horas, con picos muy claros en 9 y 10 horas. Esto sugiere una distribución más puntiaguda porque la mayoría de las mujeres parecen dormir entre 8 y 10 horas, con menos dispersión fuera de este rango.

Aunque también hay una alta concentración entre 8 y 10 horas, el pico en 9 horas es más bajo en comparación con el de las mujeres, y los valores están un poco más dispersos. Esto sugiere que el histograma de los hombres es menos puntiagudo y más aplanado o chato que el de las mujeres.

## 10. Construya una tabla cruzada entre la variable “Materia preferida en el . Colegio” versus la variable “Género”. A partir de esta tabla responda las siguientes preguntas:

In [ ]:
crosstab_Favorite_School_Subject = pd.crosstab(df.Favorite_School_Subject, df.Gender)
crosstab_Favorite_School_Subject

Gender                      Female  Male
Favorite_School_Subject                 
Art                            132    41
Computers and technology        12    43
English                         97    33
Geography                        4     7
History                         61    84
Languages                       17    12
Mathematics and statistics     130   128
Music                           55    26
Other                           40    41
Physical education              58   182
Science                         88    81
Social studies                  30    23

###  a) ¿A cuantos estudiantes hombres les gusta la historia?

Según la tabla:

- **Número de hombres que les gusta la historia**: **84**.

Entonces, **84 estudiantes hombres** prefieren la historia.

###  b) ¿Qué porcentaje de estudiantes son hombres?

In [ ]:
cuenta = df["Gender"].value_counts()
print(cuenta)
promedio_Hombres = 701*100/1425
print(f"El porcentaje estudiantes hombres es {promedio_Hombres}%")

Gender
Female    724
Male      701
Name: count, dtype: int64
El porcentaje estudiantes hombres es 49.19298245614035%


### c) ¿Qué porcentaje de estudiantes prefieren música?

In [ ]:
# Calcular el total de estudiantes que prefieren música
total_musica = crosstab_Favorite_School_Subject.loc['Music'].sum()

# Calcular el total de estudiantes
total_estudiantes = df.shape[0]

# Calcular el porcentaje de estudiantes que prefieren música
porcentaje_musica = (total_musica / total_estudiantes) * 100

print(f"El porcentaje de estudiantes que prefieren música es: {porcentaje_musica:.2f}%")


El porcentaje de estudiantes que prefieren música es: 5.68%


### d) ¿De las mujeres, qué porcentaje de estudiantes prefieren otras actividades?

Para calcular el porcentaje de mujeres que prefieren otras actividades:

1. **Número de mujeres que prefieren "Other"** (según la tabla): **40.**
2. **Total de mujeres**: **724**.

Porcentaje de mujeres que prefieren "Other” = (40 / 724) * 100

= 5.52%

Entonces, aproximadamente **5.52%** de las mujeres prefieren otras actividades.

### e) ¿De los estudiantes que prefieren Ciencias, qué porcentaje son hombres?

In [ ]:
estudiantes = crosstab_Favorite_School_Subject.loc['Science', :]
print(estudiantes)

PorcentajeHombres = (81*100)/169
print(f"El porcentaje de estudiantes hombres que prefieren Ciencias que es {PorcentajeHombres}%")

Gender
Female    88
Male      81
Name: Science, dtype: int64
El porcentaje de estudiantes hombres que prefieren Ciencias que es 47.928994082840234%


###  f) ¿Qué porcentaje son mujeres y prefieren otras actividades?

In [ ]:
# Calcular el total de mujeres que prefieren otras actividades
mujeres_other = crosstab_Favorite_School_Subject.loc['Other']['Female']

# Calcular el total de estudiantes
total_estudiantes = df.shape[0]

# Calcular el porcentaje de mujeres que prefieren otras actividades
porcentaje_mujeres_other = (mujeres_other / total_estudiantes) * 100

print(f"El porcentaje de mujeres que prefieren otras actividades es: {porcentaje_mujeres_other:.2f}%")


El porcentaje de mujeres que prefieren otras actividades es: 2.81%


### g) ¿Qué porcentaje son mujeres o prefieren música?

Para responder esta pregunta, necesitamos calcular el porcentaje de estudiantes que son mujeres o que prefieren música (incluye tanto a mujeres como a hombres que prefieren música).

1. **Número de mujeres**: **724**.
2. **Número total de hombres que prefieren música**: 26

Número de estudiantes que son mujeres o prefieren música: 724 + 26 = 750

(El numero total de hombres que prefieren musica ya está incluido en el numero de mujeres)

Ahora calculamos el porcentaje:

Porcentaje que son mujeres o prefieren música = (750 / 1425) * 100 = 52.63%

Entonces, aproximadamente **52.63%** de los estudiantes son mujeres o prefieren música.

### h) ¿De las mujeres, qué porcentaje le gusta la música o las Ciencias?

In [ ]:
Subjects = crosstab_Favorite_School_Subject.loc[['Music','Science'], 'Female']
print(Subjects)

mujeres = crosstab_Favorite_School_Subject['Female'].sum()

PorcentajeMujeres = (143*100)/mujeres
print(f"El porcentaje de estudiantes que prefieren Musica o Ciencias que son mujeres es {PorcentajeMujeres}%")

Favorite_School_Subject
Music      55
Science    88
Name: Female, dtype: int64
El porcentaje de estudiantes que prefieren Musica o Ciencias que son mujeres es 19.751381215469614%
